src/itemstackmetadata.cpp

```diff
@@ -58,11 +58,13 @@ bool ItemStackMetadata::setString(const std::string &name, const std::string &va
 	return result;
 }
 
-void ItemStackMetadata::serialize(std::ostream &os) const
+void ItemStackMetadata::serialize(std::ostream &os, bool disk) const
 {
 	std::ostringstream os2;
 	os2 << DESERIALIZE_START;
 	for (const auto &stringvar : m_stringvars) {
+		if (! disk && stringvar.first != TOOLCAP_KEY && stringvar.first != "description" && stringvar.first != "color")
```

Comment:
Missing `"palette_index"` and `"short_description"`: https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L2026